### PR TITLE
Bump to `go` `1.18`

### PR DESCRIPTION
**Pull request by jeffwidman**
From https://golang.org/dl/:
* go1.18.linux-amd64.tar.gz
* `e85278e98f57cdb150fe8409e6e5df5343ecb13cebf03a5d5ff12bd55a80264f`

### CODE REVIEW

**Comment by jeffwidman** [comment]
Still needs a test fixture or two that explicitly lists `go 1.18`.

But I'm semi-blocked on locally running `go 1.18` until https://github.com/Homebrew/homebrew-core/pull/91369 lands... based on `1.17`, it took a couple of weeks.

**Comment by jurre** [comment]
I'll chalk this up to a race condition: https://github.com/dependabot/dependabot-core/pull/4855 😄 

But let's go with your PR 👍

**Comment by jeffwidman** [comment]
>> Also need to update [dependabot/gomodules-extracted](https://github.com/dependabot/gomodules-extracted/)

> I don't think it's used anymore right?

Yep, my bad.

File: go_modules/spec/fixtures/projects/nonexisting_proxy/go.mod
Comment (jeffwidman) [inline go_modules/spec/fixtures/projects/nonexisting_proxy/go.mod:3]: @jurre any idea? Was this accident or on-purpose? Just wondering if it's supposed to be always bumped to a non-existing version of go (which means the test needs changing because there's no way this will get remembered on every bump of go).

Comment (jurre) [reply go_modules/spec/fixtures/projects/nonexisting_proxy/go.mod:3]: I'm not sure 🤔 I see that I added this file, but I don't think it's relevant for the test. I think the main thing that we need to make sure of is that we don't change the  go version specified in the go.mod file. I don't think what is in there is used, we strip it out when copying over the file: https://github.com/dependabot/dependabot-core/blob/add0e471c1920e2a726273aee7f84a37429b31c6/go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb#L115-L127

Comment (jeffwidman) [reply go_modules/spec/fixtures/projects/nonexisting_proxy/go.mod:3]: Ah, I see what this test is actually trying to test--that a non-existing proxy (`dependabot.com/nonexistingdep`) error gets handled correctly. This has nothing to do with the `go 1.18`. But should be safe enough to leave as-is (I'll remove the TODO note of course).

**Comment by jeffwidman** [comment]
The `go_modules` test failure is fixed in https://github.com/dependabot/dependabot-core/pull/4868

I broke that out separate because I think the bug is not version specific, it just
happens to only get exposed by the test suite when updating to `go 1.18`. So
keeping as two separate PR's means that if this version upgrade has to be
reverted, that change won't also be reverted.

File: go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb
Comment (mctofu) [inline go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb:165]: 🎉 Thanks for cleaning that up!

**Review by mctofu — COMMENTED** [review]
Looks good but lets rebase after merging https://github.com/dependabot/dependabot-core/pull/4868 so we can see the go tests pass?

**Review by mctofu — APPROVED** [review]
👍

**Pull request merged**
